harvester/OAIRepository.py

```diff
@@ -189,21 +198,46 @@ def unpack_oai_metadata(self, record):
 			if "bounding" in record.keys():
 				# Sometimes point data is hacked in as a bounding box
 				if record["westbc"] == record["eastbc"] and record["northbc"] == record["southbc"]:
-					record["geospatial"] = {"type": "Point", "coordinates": [[[record["northbc"][0], record["westbc"][0]]]]}
+					try:
+						record["geospatial"] = {"type": "Point", "coordinates": [[
+							[float(record["northbc"][0]), float(record["westbc"][0])]
+						]]}
+					except ValueError:
+						self.logger.error("Problem parsing geo data for record: {}".format(record["identifier"]))
+						pass
 				else:
-					record["geospatial"] = {"type": "Polygon", "coordinates": [[[record["northbc"][0], record["westbc"][0]], [record["northbc"][0], record["eastbc"][0]], [record["southbc"][0], record["westbc"][0]], [record["southbc"][0], record["eastbc"][0]]]]}
-
+					try:
+						record["geospatial"] = {"type": "Polygon", "coordinates": [[
+							[float(record["northbc"][0]), float(record["westbc"][0])],
+							[float(record["northbc"][0]), float(record["eastbc"][0])],
+							[float(record["southbc"][0]), float(record["westbc"][0])],
+							[float(record["southbc"][0]), float(record["eastbc"][0])]
+						]]}
+					except ValueError:
+						self.logger.error("Problem parsing geo data for record: {}".format(record["identifier"]))
+						pass
 
 		if self.metadataprefix.lower() == "frdr":
 			record["coverage"] = record.get("geolocationPlace")
 
 			if "geolocationPoint" in record.keys():
-				record["geospatial"] = {"type": "Point", "coordinates": [[record["geolocationPoint"][0].split()]]}
+				try:
+					record["geospatial"] = {"type": "Point", "coordinates": [
+						[float(v) for v in record["geolocationPoint"][0].split()]
```

Comment:
just curious why we need to manually recast floats here, I see we were basically doing that above, was it just an oversight before?

Reply:
If it's not a valid float, for example "18.3g" or "43.2," (note extra comma) then the float() will throw a value error. I'm trying to force the error here.

The purpose of this is to catch more user errors.  But feel free to suggest a better way if you see one?  Certainly we would also need to revisit it in the future and use locale.atof() if we're going to support French metadata entry.

I also considered just parsing what could be parsed, but then typos like "-2#3.56" would just end up being -2, instead of the more likely desired value of -23.56.  If we throw an error every time, we can notify the corresponding repo owner.

Reply:
oh! duh. I was reading it as trying to make the values compliant and wondering why it wasn't there before, not as basically adding an assertion test, which makes total sense.

Reply:
We appear to be losing some records from the BC Data Catalogue with this change though... might be dropping more than intended.

Reply:
I didn't think I changed the way the geospatial data was assembled.  Do you have an example of a record that is dropped now, that was not dropped before?